### ISSUE TITLE
`ignoreKtx(true)` doesn't work with `androidx.room:room`

### ISSUE THREAD

**Issue by BraisGabin** [autonomousapps/dependency-analysis-gradle-plugin#979]
**Plugin version**
1.24.0

**Gradle version**
8.3

**JDK version**
17

**(Optional) Android Gradle Plugin (AGP) version**
8.1.2

**(Optional) `reason` output for bugs relating to incorrect advice**
```
----------------------------------------
You asked about the dependency 'androidx.room:room-common:2.5.2'.
There is no advice regarding this dependency.
----------------------------------------

Shortest path from :asdf to androidx.room:room-common:2.5.2 for releaseCompileClasspath:
:asdf
\--- androidx.room:room-ktx:2.5.2
      \--- androidx.room:room-common:2.5.2

Shortest path from :asdf to androidx.room:room-common:2.5.2 for releaseRuntimeClasspath:
:asdf
\--- androidx.room:room-ktx:2.5.2
      \--- androidx.room:room-common:2.5.2

Shortest path from :asdf to androidx.room:room-common:2.5.2 for releaseUnitTestCompileClasspath:
:asdf
\--- androidx.room:room-ktx:2.5.2
      \--- androidx.room:room-common:2.5.2

Shortest path from :asdf to androidx.room:room-common:2.5.2 for releaseUnitTestRuntimeClasspath:
:asdf
\--- androidx.room:room-ktx:2.5.2
      \--- androidx.room:room-common:2.5.2

Source: release, main
---------------------
* Uses 8 classes, 5 of which are shown: androidx.room.Dao, androidx.room.Database, androidx.room.Entity, androidx.room.Insert, androidx.room.PrimaryKey (implies implementation).

Source: release, test
---------------------
(no usages)
```

**Describe the bug**
I'm asked to add the transitive dependency `androidx.room:room` when I have the dependency `androidx.room:room-ktx`. This seems like a regression to me. On 1.20.0 DAGP didn't ask me for that because I have `ignoreKtx(true)`. But it seems that this flag is not honored anylonger for the room case. In other cases `ignoreKtx(true)` seems to work.

**To Reproduce**
I'm not 100% sure how to reproduce it. I can create a simple project if it is needed.

**Expected behavior**
DAGP shouldn't suggest me to add this dependency.

**Additional context**
For the reason I ran `./gradlew :asdf:reason --id androidx.room:room` but because of #978 I'm getting the reason of `androidx.room:room-common` instead.

**Comment by autonomousapps** [comment]
You can provide the full coordinates (with version string), e.g. `androidx.room:room:2.5.2`

**Comment by BraisGabin** [comment]
TL;DR: I tested this on 1.25.0 and the issue is gone. DAGP doesn't ask me to add `androidx.room:room`. So closing.

---

v1.24.0:

When I run `./gradlew :asdf:reason --id androidx.room:room:2.6.0` I get this error:

```
> A failure occurred while executing com.autonomousapps.tasks.ReasonTask$ExplainDependencyAdviceAction
   > There is no dependency with coordinates 'androidx.room:room:2.6.0' in this project.
```

But in the `build-health-report.txt` I see:

```
Advice for :asdf
These transitive dependencies should be declared directly:
  implementation("androidx.room:room:2.6.0") { capabilities {
    requireCapability("androidx.room:room")
  }}
```

But this seems related with #978 and not this issue.
